src/diffusers/models/attention_processor.py

```diff
@@ -1975,6 +1975,248 @@ def __call__(self, attn: Attention, hidden_states: torch.FloatTensor, *args, **k
         return attn.processor(attn, hidden_states, *args, **kwargs)
 
 
+class IPAdapterAttnProcessor(nn.Module):
+    r"""
+    Attention processor for IP-Adapater.
+
+    Args:
+        hidden_size (`int`):
+            The hidden size of the attention layer.
+        cross_attention_dim (`int`):
+            The number of channels in the `encoder_hidden_states`.
+        text_context_len (`int`, defaults to 77):
+            The context length of the text features.
+        scale (`float`, defaults to 1.0):
+            the weight scale of image prompt.
+    """
+
+    def __init__(self, hidden_size, cross_attention_dim=None, text_context_len=77, scale=1.0):
```

Comment:
as discussed in https://github.com/tencent-ailab/IP-Adapter/pull/39, I think we should aslo use num_token instead of text_context_len to split text features and image features in IPAdapterAttnProcessor. As the users may input prompt_embeds with length more than 77.

Reply:
done